Source/DafnyCore/AST/Types/Types.cs

```diff
@@ -1101,8 +1101,8 @@ public enum AutoInitInfo { MaybeEmpty, Nonempty, CompilableValue }
                 return true;
               }
             }
-          } else if (udtSub.ResolvedClass is ClassDecl) {
-            var cl = (ClassDecl)udtSub.ResolvedClass;
+          } else if (udtSub.ResolvedClass is ClassDecl or TraitDecl) {
```

Comment:
why not `is ClassLikeDecl` ? I get a bunch more hits on `ClassDecl or TraitDecl`, so if you change one consider changing all of them.

Reply:
Done.